slices/rabbitmq-server.yaml

```diff
@@ -0,0 +1,62 @@
+package: rabbitmq-server
+
+essential:
+  - rabbitmq-server_copyright
+
+slices:
+  bins:
+    essential:
+      - adduser_bins
+      - base-files_base
+      ## erlang* and logrotate packages do not exist for i386
+      ## for all other archs, uncomment all those slices below
+      # - erlang-base_bins
+      # - erlang-crypto_libs
+      # - erlang-eldap_libs
+      # - erlang-inets_bins
+      # - erlang-mnesia_libs
+      # - erlang-os-mon_libs
+      # - erlang-parsetools_libs
+      # - erlang-public-key_libs
+      # - erlang-runtime-tools_libs
+      # - erlang-ssl_libs
+      # - erlang-syntax-tools_libs
+      # - erlang-tools_libs
+      # - erlang-xmerl_libs
+      # - logrotate_bins
+      - openssl_bins
+      - python3.12_standard
+      - rabbitmq-server_config
+      - rabbitmq-server_libs
+      - socat_bins
+    contents:
+      /usr/bin/rabbitmqadmin:
+      /usr/lib/rabbitmq/bin/rabbitmq-*:
+      /usr/lib/rabbitmq/bin/vmware-rabbitmq:
+      /usr/lib/rabbitmq/lib/rabbitmq_server-3.12.1/sbin/rabbitmq-*:
+      /usr/lib/rabbitmq/lib/rabbitmq_server-3.12.1/sbin/vmware-rabbitmq:
+      /usr/sbin/rabbitmq-*:
+
+  config:
+    contents:
+      /etc/default/rabbitmq-server:
+      /etc/init.d/rabbitmq-server:
+      /etc/logrotate.d/rabbitmq-server:
+      /etc/rabbitmq/rabbitmq-env.conf: {symlink: /usr/share/rabbitmq-env.conf}
+      /usr/lib/rabbitmq/autocomplete/*:
+      /usr/lib/systemd/system/rabbitmq-server.service:
+      /var/lib/rabbitmq/:
+      /var/lib/rabbitmq/.erlang.cookie: {text: DUMMY, mode: 0o400}
```

Comment:
```sh
  # We generate the erlang cookie by ourselves, just to make sure we don't
  # leave the job to erlang that doesn't do it with enough entropy.
  if ! [ -e /var/lib/rabbitmq/.erlang.cookie ] ; then
    OLD_UMASK=$(umask)
    umask 077; openssl rand -base64 -out /var/lib/rabbitmq/.erlang.cookie 42                                                                                                                     
    umask ${OLD_UMASK}
  else
    # This matches an Erlang generated cookie file: 20 upper case chars
    if grep -q -E '^[A-Z]{20}$' /var/lib/rabbitmq/.erlang.cookie ; then
      OLD_UMASK=$(umask)
      umask 077; openssl rand -base64 -out /var/lib/rabbitmq/.erlang.cookie 42
      umask ${OLD_UMASK}
      if [ ""$(ps --no-headers -o comm 1) = "systemd" ] ; then
        if systemctl is-active --quiet rabbitmq-server.service ; then
          systemctl restart rabbitmq-server.service
        fi  
      fi  
    fi  
  fi  
  chmod 0400 /var/lib/rabbitmq/.erlang.cookie
```

Looking at the `postinst` script, it seems like the cookie file is generated with 42 bytes of base64 data unless the cookie file is already there and contains 20 uppercase chars. I don't have enough info on how the contents of the cookie file should look like, but I think `DUMMY` is not appropriate.

Reply:
the contents of this cookie file is created by running the openssl command in rockcraft.yaml file. the contents here will be overwritten.